user_guide_src/source/general/configuration.rst

```diff
@@ -132,35 +158,53 @@ prefix, followed by a dot (.), and then the variable name itself::
     backend.db = admin
     BackEnd.db = admin
 
-Incorporating Environment Variables Into a Configuration
-========================================================
+Configuration Classes and Environment Variables
+===============================================
+
+When you instantiate a configuration class, any *namespaced* environment variables
+are considered for merging into the configuration object's properties.
+
+If the prefix of a namespaced variable exactly matches the namespace of the configuration 
+class, then the trailing part of the setting (after the dot) is treated as a configuration 
+property. If it matches an existing configuration property, the environment variable's 
+value will replace the corresponding value from the configuration file. If there is no match, 
+the configuration class properties are left unchanged. In this usage, the prefix must be 
+the full (case-sensitive) namespace of the class. 
+::
+
+    Config\App.CSRFProtection  = true    
+    Config\App.CSRFCookieName = csrf_cookie
+    Config\App.CSPEnabled = true
+
+
+.. note:: Both the namespace prefix and the property name are case-sensitive. They must exactly match the full namespace and property names as defined in the configuration class file.
 
-When you instantiate a configuration file, any namespaced environment variables
-are considered for merging into the configuration objects' properties.
+The same holds for a *short prefix*, which is a namespace using only the lowercase version of 
+the configuration class name. If the short prefix matches the class name, 
+the value will from **.env** replaces the configuration file value.
```

Comment:
I'm guessing `will` should be removed from this sentence?

Reply:
Done.